Documentation/specs/runtime-configuration-file.md

```diff
@@ -282,7 +276,54 @@ Some of the sections in the `.deps.json` file contain data used for runtime comp
 
 An application can be deployed in a "portable" deployment model. In this case, the RID-specific assets of packages are published within a folder structure that preserves the RID metadata. However, `corehost` does not use this folder structure, rather it reads data from the `.deps.json` file. Also, during deployment, the `.exe` file (`corehost` renamed) is not deployed.
 
-In a `portable` target, the package entries may have an additional `subtargets` section detailing RID-specific assets. The `corehost` application should use this data, along with the current RID and the fallback data defined in the `runtimes` section to select one **and only one** "subtarget" out of each package individually. The most specific subtarget should always be selected. In practice, this means selecting the first RID shown on the appropriate line in `runtimes`. For example, given a package containing the following subtargets:
+In the portable deployment model, the `*.runtimeConfig.json` file will contain the `targetRuntime` section:
+
+```json
+{
+    "targetRuntime": {
+        "name": "NETCore.App",
+        "version": "1.0.1"
+    }
+}
+```
+
+This data is used to locate the shared framework folder. The exact mechanics of which version are selected are defined elsewhere, but in general, it locates the shared runtime in the `shared` folder located beside it by using the relative path `shared/[targetRuntime.name]/[targetRuntime.version]`. Once it has applied any version roll-forward logic and come to a final path to the shared framework, it locates the `[targetRuntime.name].deps.json` file within that folder and loads it **first**.
+
+Next, the deps file from the application is loaded and merged into this deps file (this is conceptual, the host implementation doesn't necessary have to directly merge the data ;)). Data from the app-local deps file trumps data from the shared framework.
+
+The shared framework's deps file will also contain a `runtimes` section defining the fallback logic for all RIDs known to that shared framework. For example, a shared framework deps file installed into a Ubuntu machine may look something like the following:
+
+```json
+{
+    "targets": {
```

Comment:
Yeah. This also raises an issue during the merging where the `runtimeTarget` from the shared framework deps file should be merged with the `runtimeTarget` from the app-local deps file. It's less important to have a `runtimeTarget` in the shared framework deps file because it will only have one target, but for consistency we should have it.